app/bundles/LeadBundle/Tests/EventListener/CampaignSubscriberTest.php

```diff
@@ -0,0 +1,96 @@
+<?php
+
+/*
+ * @copyright   2016 Mautic Contributors. All rights reserved
+ * @author      Mautic, Inc.
+ *
+ * @link        https://mautic.org
+ *
+ * @license     GNU/GPLv3 http://www.gnu.org/licenses/gpl-3.0.html
+ */
+
+namespace Mautic\LeadBundle\Tests\EventListener;
+
+use Mautic\CampaignBundle\Event\CampaignExecutionEvent;
+use Mautic\CampaignBundle\Model\CampaignModel;
+use Mautic\CoreBundle\Helper\IpLookupHelper;
+use Mautic\LeadBundle\Entity\Lead;
+use Mautic\LeadBundle\EventListener\CampaignSubscriber;
+use Mautic\LeadBundle\Helper\IdentifyCompanyHelper;
+use Mautic\LeadBundle\Model\CompanyModel;
+use Mautic\LeadBundle\Model\FieldModel;
+use Mautic\LeadBundle\Model\LeadModel;
+use Mautic\LeadBundle\Model\ListModel;
+
+class CampaignSubscriberTest extends \PHPUnit_Framework_TestCase
+{
+    /** @var array */
+    private $configFrom = [
+        'companyname'       => 'Mautic',
+        'companemail'       => 'mautic@mautic.com',
+    ];
+
+    private $configTo = [
+        'companyname'       => 'Mautic2',
+        'companemail'       => 'mautic@mautic2.com',
+    ];
+
+    public function testOnCampaignTriggerActiononUpdateCompany()
+    {
+        $mockIpLookupHelper = $this->getMockBuilder(IpLookupHelper::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+
+        $mockLeadModel = $this->getMockBuilder(LeadModel::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+
+        $mockLeadFieldModel = $this->getMockBuilder(FieldModel::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+
+        $mockListModel = $this->getMockBuilder(ListModel::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+
+        $mockCompanyModel = $this->getMockBuilder(CompanyModel::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+
+        $mockCampaignModel = $this->getMockBuilder(CampaignModel::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+
+        $subscriber = new CampaignSubscriber($mockIpLookupHelper, $mockLeadModel, $mockLeadFieldModel, $mockListModel, $mockCompanyModel, $mockCampaignModel);
+
+        $lead = new Lead();
+        $lead->setId(54);
+        $mockLeadModel->expects($this->once())->method('saveEntity')->with($lead);
+
+        $helper     = new IdentifyCompanyHelper();
+        $reflection = new \ReflectionClass(IdentifyCompanyHelper::class);
+        $method     = $reflection->getMethod('identifyLeadsCompany');
+        $method->setAccessible(true);
+        list($company, $leadAdded, $companyEntity)  = $method->invokeArgs($helper, [$this->configFrom, $lead, $mockCompanyModel]);
+        $mockCompanyModel->expects($this->once())->method('addLeadToCompany')->with($companyEntity, $lead);
```

Comment:
@escopecz @alanhartless could somebody hit me with help with unit test
My tested workflow should be

- Create Lead
- Create Company from configFrom
- Add company to Lead as primary 
- Run campaign action width update contact's primary with configTo
- Check IF action results is true and if contact's primary company name is equal to configTo

Reply:
@kuzmany You did well with the unit test in the other PR. Is there still something you need help with? Also Travis checks are failing.

Reply:
@escopecz 

I am not able to finish unit test. Can you help me? I've added this part of code https://github.com/mautic/mautic/pull/5720/files/eb55b59109b698e5d6b2c9b6bb19b3a7260cf750#diff-b1aa5e9ad96d3bb9e35da9fcc121739eR38

Iam trying create contact, add company to contact and then compare if created contact has same company name as I tryied set before.

Reply:
@escopecz could you help @kuzmany here?

Reply:
`IdentifyCompanyHelper::identifyLeadsCompany` is public static method. You are trying to invoke it as protected method? This method should not be mocked. And I think it's not even possible. Let it do what it should. It will call some methods from the CompanyModel mock, so you can check what those methods will get and return.

Reply:
@escopecz Then Can I execute in test "onCampaignTriggerActionUpdateCompany" like in campaign. 
Now I change test
- create Lead
- create Company
- assign Company to Lead
- exuecute onCampaignTriggerActionUpdateCompany
- ten I want to see changed lead's primary company  

Does that test flow correct ?